sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -536,7 +536,7 @@ private[client] class Shim_v2_0 extends Shim with Logging {
         recordHiveCall()
         hive.getPartitionsByNames(table, partNames.asJava)
       } catch {
-        case ex: InvocationTargetException if ex.getCause.isInstanceOf[MetaException] =>
+        case ex: HiveException if ex.getCause.isInstanceOf[MetaException] =>
```

Comment:
Perhaps there is no test case that covers this exceptional scenario.

Reply:
yea, I think so.

Reply:
Is it possible for an `InvocationTargetException` to occur?

Reply:
I have pasted my conclusion in the PR description, please let me know if I missed something

> Directly method call won't throw `InvocationTargetException` ...